ocaml/xapi/xapi_pgpu.ml

```diff
@@ -450,8 +450,13 @@ let nvidia_vf_setup ~__context ~pf ~enable =
   let activate_vfs pci =
     match num_vfs pci with
     | None ->
-        let msg = sprintf "Can't determine number of VFs for PCI %s" pci in
-        raise (fail msg)
+        let host = Db.PCI.get_host ~__context ~self:pf in
+        let host_uuid = Db.Host.get_uuid ~__context ~self:host in
+        let device = Db.PCI.get_device_name ~__context ~self:pf in
+        error "Can't determine number of VFs for PCI %s" pci ;
+        raise
+          Api_errors.(
+            Server_error (nvidia_sriov_misconfigured, [host_uuid; device]))
```

Comment:
We should return refs in API errors rather than UUIDs.

Reply:
Will fix it. The seconds parameter is a string but I could also return something else like the PCI ref but thought this was less helpful

Reply:
Think about whether the string is useful to an API client, which might want to parse it. In this case, a name may be fine if it is just for presenting info to the user, but it won't be as unambigious as e.g. the PCI bus ID.